src/DagGenerator.cpp

```diff
@@ -379,6 +457,9 @@ static bool WriteNodes(
       BinarySegmentWriteNullPointer(node_data_seg);
     }
 
+    EmitFileSignatures(node, node_data_seg, array2_seg, str_seg);
+    EmitGlobSignatures(node, node_data_seg, array2_seg, str_seg, heap, scratch);
```

Comment:
why do these things happen an extra time now? i'm not quite following this part

Reply:
we have dag-global file and glob signatures,  but now we also have node specific ones.   so we write out both when we write out the dag.

when we run the build, we evaluate the dag global ones on startup,  and we evaluate the node specific ones after the node has executed.